lib/elixir/lib/calendar/date/range.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Date.Range do
+  @moduledoc """
+  Defines a range of dates.
+
+  A range of dates represents a discrete number of dates where
+  the first and last values are dates with matching calendars.
+
+  Ranges of dates can be either increasing (`first <= last`) or
+  decreasing (`first > last`). They are also always inclusive.
+
+  A range of dates implements the `Enumerable` protocol, which means
+  functions in the `Enum` module can be used to work with
+  ranges:
+
+      iex> range = Date.range(~D[2001-01-01], ~D[2002-01-01])
+      iex> Enum.count(range)
+      366
+      iex> Enum.member?(range, ~D[2001-02-01])
+      true
+      iex> Enum.reduce(range, 0, fn(_date, acc) -> acc - 1 end)
+      -366
+
+  """
+
+  defstruct [:first, :last, :kind, :first_rata_die, :last_rata_die]
+
+  def new(%{calendar: fc} = first, %{calendar: lc} = last) when fc == lc do
```

Comment:
No shortcut variables such as `fc` or `lc`. Call it `first_calendar`, `last_calendar`, etc.